language-server/src/main/java/org/sonarlint/languageserver/SonarLintLanguageServer.java

```diff
@@ -153,12 +177,12 @@ public SonarLintLanguageServer(InputStream inputStream, OutputStream outputStrea
     backgroundProcess = launcher.startListening();
   }
 
-  public static SonarLintLanguageServer bySocket(int port, Collection<URL> analyzers) throws IOException {
+  static SonarLintLanguageServer bySocket(int port, Collection<URL> analyzers) throws IOException {
     Socket socket = new Socket("localhost", port);
     return new SonarLintLanguageServer(socket.getInputStream(), socket.getOutputStream(), analyzers);
   }
 
-  private static class UserSettings {
+  private class UserSettings {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make this a "static" inner class. ([squid:S2694](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2694))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=192&issues=AWM3TWeR_vaxgEW1HRnk&open=AWM3TWeR_vaxgEW1HRnk)

[ISSUE_KEY]: AWM3TWeR_vaxgEW1HRnk